sdk/core/src/retrieve_tx_on_network.rs

```diff
@@ -35,27 +38,24 @@ pub async fn retrieve_tx_on_network<GatewayProxy: GatewayAsyncService>(
                             "Transaction retrieved successfully, with status {}: {:#?}",
                             status, transaction_info_with_results
                         );
-                        return transaction_info_with_results;
+                        return (transaction_info_with_results, ReturnCode::Success);
                     },
                     "fail" => {
-                        // status failed and no reason means invalid transaction
-                        if reason.is_empty() {
-                            info!("Transaction failed. Invalid transaction: {tx_hash}");
-                            panic!("Transaction failed. Invalid transaction: {tx_hash}");
-                        }
-
-                        let result = parse_reason(&reason);
-
-                        match result {
-                            Ok((code, err)) => {
-                                info!("Transaction failed. Code: {code}, message: {err}");
-                                panic!("Transaction failed. Code: {code}, message: {err}")
-                            },
-                            Err(err) => {
-                                info!("Reason parsing error for failed transaction: {err}");
-                                panic!("Reason parsing error for failed transaction: {err}")
-                            },
-                        }
+                        let (error_code, error_message) = parse_reason(&reason);
```

Comment:
Currently, `LogData` field contains a **signalError event**. The second topic of the event contains the error message of the transaction. Whenever there is an empty reason, the event's topic is not altered. For example, the next two transactions have an empty reason: 
- https://devnet-explorer.multiversx.com/transactions/d9db8aa162e1ad70e962b5152e8fd182144aa565ff8469dfcf6d2d36526f08b0
- https://devnet-explorer.multiversx.com/transactions/c0696f3290c0ef6834cb40525f53b4f5c3196aa9a0d7050d85454b5b155543dd

I find it too vague to mark both transactions as `"invalid transaction"` since `TransactionOnNetwork` contains the reason why the transaction failed in the log data.